python/lsst/multiprofit/sourceconfig.py

```diff
@@ -0,0 +1,385 @@
+        default={},
+    )
+    components_sersic = pexConfig.ConfigDictField[str, SersicComponentConfig](
+        doc="Sersic Components in the component mixture",
+        optional=False,
+        default={},
+    )
+    is_fractional = pexConfig.Field[bool](doc="Whether the integral_model is fractional", default=False)
+    transform_fluxfrac_name = pexConfig.Field[str](
+        doc="The name of the reference transform for flux parameters",
+        default="logit_fluxfrac",
+        optional=True,
+    )
+    transform_flux_name = pexConfig.Field[str](
+        doc="The name of the reference transform for flux parameters",
+        default="log10",
+        optional=True,
+    )
+
+    @staticmethod
+    def format_label(label: str, name_component: str) -> str:
+        return string.Template(label).safe_substitute(name_component=name_component)
+
+    @staticmethod
+    def get_integral_label_default() -> str:
+        return "comp: ${name_component} " + EllipticalComponentConfig.get_integral_label_default()
+
+    def get_component_configs(self) -> ComponentConfigs:
+        component_configs: ComponentConfigs = dict(self.components_gauss)
+        for name, component in self.components_sersic.items():
+            component_configs[name] = component
+        return component_configs
+
+    @staticmethod
+    def get_fluxes_default(
+        channels: tuple[g2f.Channel],
+        component_configs: ComponentConfigs,
+        is_fractional: bool = False,
+    ) -> list[Fluxes]:
+        if len(component_configs) == 0:
+            raise ValueError("Must provide at least one ComponentConfig")
+        fluxes = []
+        component_configs_iter = tuple(component_configs.values())[: len(component_configs) - is_fractional]
+        for idx, component_config in enumerate(component_configs_iter):
+            if is_fractional:
+                if idx == 0:
+                    value = component_config.flux.value_initial
+                    fluxes.append({channel: value for channel in channels})
+                value = component_config.fluxfrac.value_initial
+                fluxes.append({channel: value for channel in channels})
+            else:
+                value = component_config.flux.value_initial
+                fluxes.append({channel: value for channel in channels})
+        return fluxes
+
+    def make_components(
+        self,
+        component_fluxes: list[Fluxes],
+        label_integral: str | None = None,
+    ) -> tuple[list[g2f.Component], list[g2f.Prior]]:
+        """Make a list of gauss2d.fit.Component from this configuration.
+
+        Parameters
+        ----------
+        component_fluxes
+            A list of Fluxes to populate an appropriate
+            `gauss2d.fit.IntegralModel` with.
+            If self.is_fractional, the first item in the list must be
+            total fluxes while the remainder are fractions (the final
+            fraction is always fixed at 1.0 and must not be provided).
```

Comment:
Besides needing to make some methods abstract, splitting the classes would require making any configs that use this (right now just `SourceConfig`) use a `RegistryField`, since `ConfigField` doesn't support derived classes. That's not a dealbreaker but slightly annoying. So I'm reluctant to refactor the one-class design until it actually causes problems, at which point (if I remember) I'll admit you were right all along.

It's true that the current design makes it somewhat tricky to get the `component_group_fluxes` kwarg correct when making a fractional model (which is pretty much exclusively for PSF models), but that mostly ends up landing on higher tier classes within MultiProFit rather than on users, so I'm ok with it. Also, the main benefit of this design is that when `SourceConfig` calls `make_components` it doesn't need to know or care whether the component is fractional or not.